com.unity.render-pipelines.universal/ShaderLibrary/ShaderTypes.cs

```diff
@@ -12,14 +12,15 @@ public struct LightData
             public Vector4 occlusionProbeChannels;
         }
 
-        [GenerateHLSL(PackingRules.Exact, false)]
-        public struct ShadowData
+        //[GenerateHLSL(PackingRules.Exact, false)]
+        // Currently we do not use this struct in the hlsl side, because worldToShadowMatrix and shadowParams must be stored in arrays of different sizes
+        public struct ShadowData // ShadowSliceData
```

Comment:
Thanks for the recommendation, I did that in revision 36c3908

Reply:
Ahh, this is a public struct in a public static class so this is a breaking change.
We need to move it to Deprecated.cs and mark it as obsolete.

Reply:
OK, I moved the `ShaderInput.ShadowData` into Deprecated.cs with revision 5664126.
I also moved the hlsl-side declaration to Deprecated.hlsl.

At first I tried to move `ShaderInput.ShadowData` into com.unity.render-pipelines.universal/Runtime/Deprecated.cs but this caused type conflicts between assemblies because `ShaderInput.ShadowData` is in assembly `Unity.RenderPipelines.Universal.ShaderLibrary` while com.unity.render-pipelines.universal/Runtime/Deprecated.cs is in assembly `Unity.RenderPipelines.Universal.Runtime`.
Moving `ShaderInput.ShadowData` into a new com.unity.render-pipelines.universal/ShaderLibrary/Deprecated.cs file included in the same `Unity.RenderPipelines.Universal.ShaderLibrary` assembly, solved the issues.